README.md

```diff
@@ -1,2 +1,73 @@
-# scraped_page
+# ScrapedPage
+
 Write declarative scrapers in Ruby
+
+## Usage
+
+Create a subclass of `ScrapedPage` for each _type_ of page you wish to scrape.
+
+For example if you were scraping a list of people you might have a
+`PeopleListPage` class for the list page and a `PersonPage` class for an
+individual person's page.
+
+```ruby
+require 'scraped_page'
+
+class ExamplePage < ScrapedPage
+  field :title do
+    noko.at_css('h1').text
+  end
+
+  field :more_information do
+    noko.at_css('a')[:href]
+  end
+end
+```
+
+Then you can create a new instance and pass in a url
+
+```ruby
+page = ExamplePage.new(url: 'http://example.com')
+
+page.title
+# => "Example Domain"
+
+page.more_information
+# => "http://www.iana.org/domains/reserved"
+
+page.to_h
+# => { :title => "Example Domain", :more_information => "http://www.iana.org/domains/reserved" }
+```
+
+## Archiving scraped pages
+
+The default strategy for retrieving pages is `ScrapedPage::Strategy::LiveRequest`. If you'd also like to archive a copy of the page you're scraping into a git branch, you can pass a `:strategy` option to the constructor:
+
+```ruby
+ExamplePage.new(url: 'http://example.com', strategy: ScrapedPage::Strategy::LiveRequestArchive.new)
+```
+
+This will use the [`scraped_page_archive`](https://github.com/everypolitician/scraped_page_archive)
+gem to store a copy of the pages you scrape in a git branch in your scraper's repo.
+
+## Custom strategies
+
+If for some reason you can't scrape a site using one of the built in strategies
+then you can provide your own strategy.
+
+A strategy is an object that implements a `response` method which returns a
+`Hash` with at least a `:body` key and optionally `:status` and `:headers` keys.
+
+```ruby
+class FilesystemStrategy
+  def response(url)
+    { body: File.read(Digest::SHA1.hexdigest(url) + '.html') }
```

Comment:
I've changed the strategy entry point method to `response` rather than `get`. As you say a noun works better and is clearer here.

> Well, the current documentation of "A strategy is an object that responds to a get method and returns a ScrapedPage::Response" raises quite a few questions along how I know what a ScrapedPage::Response is and how to generate one etc. So either we need to document that more, or change what it returns. I don't really understand deeply enough what the Response class is/does atm, so don't have much a view on that yet…

Yep, I think I was exposing too much of the guts of the library. I've changed this to just be a `Hash` for now.